### ISSUE TITLE
Drawing incorrectly with rows of different sizes

### ISSUE THREAD

**Issue by dbaio** [p-gen/smenu#5]
This is happening with version 0.9.10
OS FreeBSD.

<img width="400" alt="smenu_0 9 9" src="https://user-images.githubusercontent.com/2529335/34189704-f7db161a-e523-11e7-9bd5-8fbd0a91e98a.png">

<img width="467" alt="smenu_0 9 10" src="https://user-images.githubusercontent.com/2529335/34189708-ff73c782-e523-11e7-8419-98fbc7012812.png">

[smenu_list1.txt](https://gist.github.com/dbaio/8f54ac489557ee165995cd24c10f5a9a)
[smenu_list2.txt](https://gist.github.com/dbaio/0f1168d090e117dd2aa84b916f50a89d)

**Comment by p-gen** [comment]
I cannot reproduce this behavhiour on Linux, the outputs are the same as yours for version 0.9.9.

Dis you try with another terminal emulator (xterm by ex.)?

Can you join a tar containing your files so that I can be sure that they have not been altered?

Can you also give me the version of FreeBSD and of the libraries used so that I can try to reproduce your environment?

**Comment by dbaio** [comment]
`TERM=xterm-256color`
`FreeBSD current 1200053`

I think I found the problem, I will try it on FreeBSD 10.4 and 11 as well.

The problem was that `smenu` was linking with `ncurses` from the base system (NCURSES_VERSION "5.9").

```
$ ldd /usr/local/bin/smenu
/usr/local/bin/smenu:
        libncurses.so.8 => /lib/libncurses.so.8 (0x800839000)
        libc.so.7 => /lib/libc.so.7 (0x800a8e000)
```

Linking with a new version fixed this (NCURSES VERSION "6.0").

```
$ ldd /usr/local/bin/smenu
/usr/local/bin/smenu:
        libtinfo.so.6 => /usr/local/lib/libtinfo.so.6 (0x800839000)
        libc.so.7 => /lib/libc.so.7 (0x800a74000)
```

Regards.

**Comment by p-gen** [comment]
Great,

Notice that smenu only needs libtinfo if it is available and not a part or (n)curses.

Tell me if it is also fixed in FreeBSD 10.4 and 11. I'll close this issue if it is the case.

Regards.

**Comment by dbaio** [comment]
Hi, closing this issue, it worked.

There is an old bug to separate libncurses and libtinfo in FreeBSD base https://bugs.freebsd.org/bugzilla/show_bug.cgi?id=197317

I am building now with ncurses from ports and I'll push this update in FreeBSD.

Thanks for your time.
